### PR TITLE
Make it possible to render a GDevelop scene in VR

### PR DESCRIPTION
**Pull request by arthuro555**
These changes allow a VR extension (in the works) to function. 

It adds:

- A method to be able to render the scene without PIXI or resetting the WebGL state (necessary for the image to be rendered to the headset)
- Disabling "normal" rendering on step when a WebXR session is in progress (reason why we are not just calling the method mentioned above explained in a code comment)

### CODE REVIEW

File: GDJS/Runtime/pixi-renderers/runtimescene-pixi-renderer.ts [hunk 1]
```diff
@@ -267,6 +280,64 @@ namespace gdjs {
...
+
+        const runtimeLayerRenderer = runtimeLayer.getRenderer();
+        const runtimeLayerRenderingType = runtimeLayer.getRenderingType();
+        if (
+          runtimeLayerRenderingType === gdjs.RuntimeLayerRenderingType.TWO_D ||
+          !runtimeLayerRenderer.has3DObjects()
+        )
+          continue;
+
+        // Render a layer with 3D rendering, and possibly some 2D rendering too.
```
Comment (4ian) [inline GDJS/Runtime/pixi-renderers/runtimescene-pixi-renderer.ts:314]:
```suggestion
        // Render a layer with 3D rendering, and no 2D rendering at all for VR.
```

File: GDJS/Runtime/pixi-renderers/runtimescene-pixi-renderer.ts [hunk 2]
```diff
@@ -267,6 +280,64 @@ namespace gdjs {
...
+     * as WebXR modifies the WebGL state in a way that resetting it will cause an improper render
+     * that will lead to a black screen being displayed in VR mode.
+     */
+    renderForVR() {
+      const runtimeGameRenderer = this._runtimeGameRenderer;
+      if (!runtimeGameRenderer) return;
+
+      const threeRenderer = this._threeRenderer;
+      // VR rendering relies on ThreeJS
+      if (!threeRenderer) return;
```
Comment (4ian) [inline GDJS/Runtime/pixi-renderers/runtimescene-pixi-renderer.ts:298]: Would be I think worth to just throw an Error?

File: GDJS/Runtime/pixi-renderers/runtimescene-pixi-renderer.ts [hunk 3]
```diff
@@ -267,6 +280,64 @@ namespace gdjs {
       // console.log(this._layerRenderingMetrics);
     }
 
+    /**
+     * In VR, only 3D elements can be rendered, 2D cannot.
+     * This rendering method skips over all 2D layers and elements, and simply renders the 3D content.
+     * This method is to be called by the XRSession's requestAnimationFrame for rendering to
```
Comment (4ian) [inline GDJS/Runtime/pixi-renderers/runtimescene-pixi-renderer.ts:286]: unfinished sentence?

**Review by 4ian — APPROVED** [review]

**Comment by 4ian** [comment]
Nice! Is there a need to do a writeup/doc on the wiki for this?

**Comment by arthuro555** [comment]
I'll probably be writing up a wiki page for the extension once I finish & submit the extension that uses this.

**Comment by 4ian** [comment]
Cool! We can also add a link to it on the website features page :)

**Pull request merged**